channel.py

```diff
@@ -287,10 +300,21 @@ def recv_mb(self):
             The list of numpy arrays for the minibatch
 
         """
-        headers = self.asocket.recv_json()
+        socks = dict(self.apoller.poll(self._socket_timout))
+        if socks:
+            if socks.get(self.asocket) == zmq.POLLIN:
+                headers = self.asocket.recv_json()
+        else:
+            raise Exception("Batch socket: recv_json timeout")
+
         arrays = []
         for header in headers:
-            data = self.asocket.recv(copy=False)
+
+            if socks and socks.get(self.asocket) == zmq.POLLIN:
+                data = self.asocket.recv(copy=False)
+            else:
+                raise Exception("Batch socket: recv timeout")
+
```

Comment:
Done!

On Thu, Jan 7, 2016 at 2:53 PM, abergeron <notifications@github.com> wrote:

> In channel.py
> <https://github.com/mila-udem/platoon/pull/13#discussion_r49117582>:
>
> >          arrays = []
> >          for header in headers:
> > -            data = self.asocket.recv(copy=False)
> > +
> > +            if socks and socks.get(self.asocket) == zmq.POLLIN:
> > +                data = self.asocket.recv(copy=False)
> > +            else:
> > +                raise Exception("Batch socket: recv timeout")
> > +
>
> No, because multipart messages (which this is) are either received
> completely or not at all.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/platoon/pull/13/files#r49117582>.
>